### PR TITLE
Feature/metal support

### PR DESCRIPTION
**Pull request by sdk-banuba**
- Add METAL support to OEP-macos
- Remove OpenGL things
- Update example to SDK v1.1.0

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -97,12 +97,11 @@ set_target_properties(${APP_NAME} PROPERTIES
 )
 
 target_link_libraries(${APP_NAME}
+    "-framework ${BANUBA_OEP_FRAMEWORK}"
+    bnb_effect_player
```
Comment (vadimVoloshanov) [inline CMakeLists.txt:101]: Must be part of BANUBA_OEP_FRAMEWORK

File: CMakeLists.txt [hunk 2]
```diff
@@ -97,12 +97,11 @@ set_target_properties(${APP_NAME} PROPERTIES
...
     "-framework Accelerate"
     "-framework Cocoa"
     "-framework Foundation"
-    "-framework OpenGL"
-    "-framework BanubaEffectPlayer"
```
Comment (vadimVoloshanov) [inline CMakeLists.txt:104]: The main framework banuba_sdk, must it be included, or does it work differently in the scene?

File: offscreen_render_target/include/offscreen_render_target.h [hunk 3]
```diff
@@ -3,71 +3,62 @@
...
+    EPOrientationAngles180,
+    EPOrientationAngles270
+};
+
 namespace bnb
 {
-    class ort_frame_surface_handler;
 
-    class offscreen_render_target : public interfaces::offscreen_render_target
+    class offscreen_renderer : public interfaces::offscreen_render_target
```
[the remark this answers is not in the document]
Comment (sdk-banuba) [reply offscreen_render_target/include/offscreen_render_target.h]: Renaming canceled

File: oep_framework/oep/BNBCameraOrientation.h [hunk 4]
```diff
@@ -0,0 +1,12 @@
+// AUTOGENERATED FILE - DO NOT MODIFY!
```
Comment (sdk-banuba) [inline oep_framework/oep/BNBCameraOrientation.h:1]: Why do you need this file here if it is already present in BanubaEffectPlayer.framework

File: oep_framework/oep/BNBFullImageData.mm [hunk 5]
```diff
@@ -0,0 +1,180 @@
+#import "BNBFullImageData.h"
```
Comment (sdk-banuba) [inline oep_framework/oep/BNBFullImageData.mm:1]: Why not to use it from BanubaEffectPlayer.framework?

File: offscreen_render_target/include/offscreen_render_target.h [hunk 6]
```diff
@@ -3,71 +3,62 @@
...
-#include "program.hpp"
-
-#include <glad/glad.h>
-
+#import <Metal/Metal.h>
+#import <MetalKit/MetalKit.h>
 #import <CoreMedia/CoreMedia.h>
 
+typedef NS_ENUM(NSUInteger, EPOrientation) {
+    EPOrientationAngles0,
```
Comment (sdk-banuba) [inline offscreen_render_target/include/offscreen_render_target.h:13]: Without BNB prefix? It's not OK

File: resources/OEPShaders.metal [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    constexpr sampler samplr(s_address::clamp_to_edge,
+                             t_address::clamp_to_edge,
+                             r_address::clamp_to_edge,
+                             min_filter::linear,
+                             mag_filter::linear
+                             );
+    return half4(textureRGBA.sample(samplr, in.tex_coord).bgra);
+}
+
+} // namespace BNBOEPShaders
```
Comment (sdk-banuba) [inline resources/OEPShaders.metal:81]: Do we need  resources/OEPShaders.metallib ?

**Review by vadimVoloshanov — APPROVED** [review]

**Pull request merged**